app/api/views/briefs.py

```diff
@@ -109,19 +115,177 @@ def domain(email):
     return supplier, brief
 
 
+@api.route('/brief/rfx', methods=['POST'])
+@login_required
+@role_required('buyer')
+def create_rfx_brief():
+    if current_user.role != 'buyer':
+        return forbidden('Unauthorised to create a brief')
+    try:
+        lot = Lot.query.filter(Lot.slug == 'rfx').first()
+        framework = Framework.query.filter(Framework.slug == 'digital-marketplace').first()
+        user = User.query.get(current_user.id)
+        brief = Brief(
+            users=[user],
+            framework=framework,
+            lot=lot,
+            data={}
+        )
+        db.session.add(brief)
+        db.session.commit()
+    except Exception as e:
+        rollbar.report_exc_info()
+        return jsonify(message=e.message), 400
+
+    return jsonify(brief.serialize(with_users=False))
+
+
 @api.route('/brief/<int:brief_id>', methods=["GET"])
 def get_brief(brief_id):
     brief = Brief.query.filter(
         Brief.id == brief_id
-    ).first_or_404()
-    if brief.status == 'draft':
+    ).first()
+    if not brief:
+        not_found("No brief for id '%s' found" % (brief_id))
+
+    is_buyer = False
+    is_brief_owner = False
+    brief_user_ids = [user.id for user in brief.users]
+    if hasattr(current_user, 'role') and current_user.role == 'buyer':
```

Comment:
why is the attribute check needed? is that always part of current_user?
if it is really needed it is missing in update_brief

Reply:
Anonymous users don't have a `role` attribute. `update_brief` has a `role_required()` decorator so anonymous users don't get in. The brief GET is publicly accessible so needs anonymous user access.